op-node/rollup/derive/pipeline.go

```diff
@@ -93,13 +92,17 @@ func NewDerivationPipeline(log log.Logger, rollupCfg *rollup.Config, l1Fetcher L
 	// Reset from ResetEngine then up from L1 Traversal. The stages do not talk to each other during
 	// the ResetEngine, but after the ResetEngine, this is the order in which the stages could talk to each other.
 	// Note: The ResetEngine is the only reset that can fail.
-	stages := []ResettableStage{l1Traversal, l1Src, plasma, frameQueue, bank, chInReader, batchQueue, attributesQueue}
+	var stages []ResettableStage
+	if plasma != nil {
```

Comment:
`plasma` is also being passed into `NewDataSourceFactory` above. It should only be used when plasma is enabled but I'm getting pretty nervous about passing `nil` around so much. I'd agree that instantiating plasma and passing it around when it should be disabled is a bad pattern, but I think fixing that needs a bigger refactor.

@protolambda has been doing a lot of work to make these kinds of integrations into the pipeline more pluggable and less intrusive. Proto - any preferences here? And does your current plans to refactor things extend to this part of the plasma integration?

Reply:
I checked that the `plasma` passed into `NewDataSourceFactory` is only used when plasma is enabled: [source](https://github.com/ethereum-optimism/optimism/blob/3eb986192d22d2ef817bd758b04da90623734655/op-node/rollup/derive/data_source.go#L79)

Reply:
Yeah I'm more concerned with future changes potentially introducing `nil` dereferences.  Typically we've used no-op implementations rather than `nil` in the derivation pipeline.  I'm not strongly against this, just seems clear that the way plasma is integrated here isn't ideal and will need broader changes and I'm keen to avoid adding more unexpected conditions when we haven't done the real work required to make the integration clean.

Reply:
It seems the main difference in PoV of treating optional features is whether to use `nil` or no-op implementations. Personally I prefer `nil` to no-op implementations for optional features that was not enabled.